Core/src/org/sleuthkit/autopsy/centralrepository/datamodel/CorrelationDataSource.java

```diff
@@ -173,4 +221,72 @@ public Long getDataSourceObjectID() {
     public String getName() {
         return name;
     }
+    
+    /**
+     * @return the MD5 hash value
+     */
+    public String getMd5() {
+        return md5Hash;
+    }
+    
+    /**
+     * Set the MD5 hash value and persist to the Central Repository if available.
+     * 
+     * @param md5Hash The MD5 hash value.
+     * 
+     * @exception EamDbException If there's an issue updating the Central
+     *                           Repository.
+     */
+    public void setMd5(String md5Hash) throws EamDbException {
+        this.md5Hash = md5Hash;
+        
+        boolean useCR = EamDb.isEnabled();
+        if (useCR) {
+            EamDb.getInstance().updateDataSourceMd5Hash(this);
+        }
```

Comment:
It isn't necessary based on the scenarios under which this is utilized. Since these setters are unconventional in comparison to the others found on the CR side of the code, I felt it appropriate to take precaution and make sure it was enabled. We have this check in the 'fromTSKDataSource()' method as well, although the difference here perhaps is the calling code itself doesn't always do the check, so we have a check built in.

Anyway, I have removed this as it should never be false. However, I do think a check for a -1 object ID makes a lot of sense. That is what I have put here instead. Concur?